src/app/App.jsx

```diff
@@ -92,73 +89,77 @@ class App extends React.Component {
       this.setState({ dir: nextProps.defaultDir });
     }
+
+    this.utilityConfig = this.setupUtilityConfig(nextProps.utilityConfig);
   }
 
-  handleBidiChange(e) {
-    document.getElementsByTagName('html')[0].setAttribute('dir', e.currentTarget.id);
-    this.setState({ dir: e.currentTarget.id });
+  setupUtilityConfig(utilityConfig) {
+    return ConfigureUtilities.addCallbackFunctions(
+      utilityConfig,
+      {
+        Theme: { onChange: this.handleThemeChange },
+        Locale: { onChange: this.handleLocaleChange },
+        Bidi: { onChange: this.handleBidiChange },
+      });
   }
 
-  handleLocaleChange(e) {
-    document.getElementsByTagName('html')[0].setAttribute('lang', e.currentTarget.id);
-    this.setState({ locale: e.currentTarget.id });
+  handleBidiChange(key) {
+    document.getElementsByTagName('html')[0].setAttribute('dir', key);
+    this.setState({ dir: key });
   }
 
-  handleThemeChange(e) {
-    this.setState({ theme: e.currentTarget.id });
+  handleLocaleChange(key) {
+    document.getElementsByTagName('html')[0].setAttribute('lang', key);
+    this.setState({ locale: key });
   }
 
-  render() {
-    let appLogo;
-    if (this.props.appLogoSrc) {
-      appLogo = (<Image variant="rounded" src={this.props.appLogoSrc} height="26px" width="26px" isFluid />);
-    }
+  handleThemeChange(key) {
+    this.setState({ theme: key });
+  }
 
-    let applicationHeader;
-    if (!matchPath(this.props.location.pathname, '/raw/tests')) {
-      applicationHeader = (
-        <ApplicationHeader
-          title={this.props.appTitle}
-          logo={appLogo}
-          locale={this.state.locale}
-          locales={this.props.locales}
-          onLocaleChange={this.handleLocaleChange}
-          hideBidiUtility={this.props.hideBidiUtility}
-          dir={this.state.dir}
-          onDirChange={this.handleBidiChange}
-          theme={this.state.theme}
-          themes={Object.keys(this.props.themes)}
-          onThemeChange={this.handleThemeChange}
-          navigation={matchPath(this.props.location.pathname, this.props.rootPath) ? this.props.navigation : undefined}
-        />
-      );
-    }
+  render() {
+    const { nameConfig, location, routingConfig, navigationItems, indexPath, extensions } = this.props;
+    const { theme, locale, dir } = this.state;
+    this.utilityConfig = ConfigureUtilities.updateSelectedItems(this.utilityConfig, theme, locale, dir);
 
     return (
-      <ThemeProvider id="site" themeName={this.props.themes[this.state.theme]} isGlobalTheme>
-        <Base className="base" locale={this.state.locale}>
-          <NavigationLayout
-            header={applicationHeader}
-            menuText="Menu"
-            indexPath={this.props.navigation && this.props.navigation.index}
-            config={this.props.routeConfig}
-          />
+      <ThemeProvider id="site" themeName={appConfig.themes[theme]} isGlobalTheme>
```

Comment:
`themeName={appConfig.themes[theme]} ` I think we lost the passing of appConfig/themes in the props and we are now always using the default theme as the global theme. Consumer theme defines their own theme as the default